File: src/main/java/de/scandio/e4/worker/abstracts/WebConfluence.java
package de.scandio.e4.worker.abstracts;

import de.scandio.e4.worker.interfaces.WebClient;
import org.openqa.selenium.WebDriver;

public class WebConfluence implements WebClient {

	private WebDriver driver;

	public WebConfluence(WebDriver driver) {
		this.driver = driver;
	}

	@Override
	public WebDriver getDriver() {
		return driver;
	}

}

File: src/main/kotlin/de/scandio/e4/enjoy/WebConfluence.kt
package de.scandio.e4.enjoy

import org.openqa.selenium.By
import org.openqa.selenium.WebDriver
import java.net.URI
import java.net.URLEncoder

class WebConfluence (
    val driver: WebDriver,
    val base: URI,
    private val adminPassword: String
    ) {

        fun goToLogin(): LoginPage {
            navigateTo("login.jsp")
            return LoginPage(driver)
        }

        fun goToSystemInfo() {
            navigateTo("secure/admin/ViewSystemInfo.jspa")
        }

        fun navigateTo(path: String) {
            driver.navigate().to(base.resolve(path).toURL())
        }
}